From ecf539277191ec57b55b67232287e1c588e4bb6c Mon Sep 17 00:00:00 2001
From: Antonio Terceiro <terceiro@debian.org>
Date: Mon, 4 Jan 2016 18:22:20 -0200
Subject: [PATCH] adt-build-{lxc,lxd}: make sure testbed setup is done

the scripts in setup-commands/* are installed as data and not as proper
binaries, what makes adt-build-{lxc,lxd} not run any setup, producing
mostly broken containers.

Since the setup script is always piped into a shell instead of executed
directly, it is probably not useful to install it with the execution
bits on anyway.
---
 tools/adt-build-lxc | 2 +-
 tools/adt-build-lxd | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/adt-build-lxc b/tools/adt-build-lxc
index 89a8750..0247f2c 100755
--- a/tools/adt-build-lxc
+++ b/tools/adt-build-lxc
@@ -113,7 +113,7 @@ setup() {
     # find setup-testbed script
     for script in $(dirname $(dirname "$0"))/setup-commands/setup-testbed \
                   /usr/share/autopkgtest/setup-commands/setup-testbed; do
-        if [ -x "$script" ]; then
+        if [ -r "$script" ]; then
             echo "Running setup script $script..."
             cat "$script" | lxc-attach --name=$1 sh
             break
diff --git a/tools/adt-build-lxd b/tools/adt-build-lxd
index f525215..22e6343 100755
--- a/tools/adt-build-lxd
+++ b/tools/adt-build-lxd
@@ -82,7 +82,7 @@ setup() {
     # find setup-testbed script
     for script in $(dirname $(dirname "$0"))/setup-commands/setup-testbed \
                   /usr/share/autopkgtest/setup-commands/setup-testbed; do
-        if [ -x "$script" ]; then
+        if [ -r "$script" ]; then
             echo "Running setup script $script..."
             lxc exec "$CONTAINER" -- sh < "$script"
             break
-- 
2.6.4

